### PR TITLE
Timer: reduce cpu load

### PR DESCRIPTION
**Pull request by danimoh**
Reduce cpu load by running code just at the required interval.

I will also performance improvements to the Keyguard PR.

### CODE REVIEW

**Comment by sisou** [comment]
Was the RAF running continuously before? And now only during an animation, which in turn only runs on an interval?

**Comment by danimoh** [comment]
A rAF was running continuously before but only rerendering when animating the radius or after `_updateInterval` time passed.
This has now been decoupled into an interval and a rAF.
Repaints are mainly handled by the interval unless the radius is being animated which is done with a rAF that runs only during the animation.

The interval in the expanded timer state at the size we use in the Hub and a timeout of 15 minutes runs approximately every 4s.

File: src/components/Timer.vue [hunk 1]
```diff
@@ -212,44 +221,61 @@ class Timer extends Vue {
...
-            if (!this.sampledTime || sampledTime - this.sampledTime >= this._updateInterval
-                || !this.radius.finished) { // animating radius
-                this.sampledTime = sampledTime;
-                this.fullCircleLength = 2 * Math.PI * this.radius.currentValue;
-            }
+        clearInterval(this.updateIntervalId);
+        this.updateIntervalId = window.setInterval(() => {
+            this._rerender();
+            if (this._timeLeft !== 0) return;
+            clearInterval(this.updateIntervalId);
```
Comment (sisou) [inline src/components/Timer.vue:253]:
```suggestion
            if (this._timeLeft === 0) clearInterval(this.updateIntervalId);
```

Comment (danimoh) [reply src/components/Timer.vue:253]: We usually only have `return`s without opening a block.

Comment (sisou) [reply src/components/Timer.vue:253]: Well, you can put a block around it :stuck_out_tongue: I wanted to change it to not return early, but instead make the `clearInterval` the result of the condition. I was first confused why you are clearing the interval _inside_ the interval, and only later saw that you return early in the regular case. I think early returns should be reserved for exceptions, errors, not the expected flow.

**Pull request merged**